tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -300,8 +300,13 @@ def _make_builder(self, config=None):
   @test_utils.run_in_graph_and_eager_modes()
   def test_download_and_prepare_as_dataset(self):
     # If configs specified, ensure they are all valid
+    test_config_names = []  # extract names of all builder configs to test
+
     if self.BUILDER_CONFIG_NAMES_TO_TEST:
       for config in self.BUILDER_CONFIG_NAMES_TO_TEST:  # pylint: disable=not-an-iterable
+        if isinstance(config, dataset_builder.BuilderConfig):
+          config = config.name
+        test_config_names.append(config)
         assert config in self.builder.builder_configs, (
```

Comment:
I would rather do something like:

```python
configs_to_test = []
for config in self.BUILDER_CONFIG_NAMES_TO_TEST:
  if isinstance(config, dataset_builder.BuilderConfig):
    configs_to_test.append(config)
  elif config in self.builder.builder_configs:
    configs_to_test.append(self.builder.builder_configs[config])
  else:
    raise AssertionError(f'Invalid config {config}. Available: {list(self.builder.builder_configs)}')
```

Then use `configs_to_test` bellow, rather than`configs`